libs/pages/application/src/lib/feature/page-settings-configure-feature/page-settings-configure-feature.tsx

```diff
@@ -0,0 +1,157 @@
+import { JobResponse } from 'qovery-typescript-axios'
+import { useEffect, useState } from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
+import { useDispatch, useSelector } from 'react-redux'
+import { useParams } from 'react-router-dom'
+import { editApplication, postApplicationActionsRestart, selectApplicationById } from '@qovery/domains/application'
+import { ServiceTypeEnum, getServiceType, isCronJob, isLifeCycleJob } from '@qovery/shared/enums'
+import { ApplicationEntity, JobConfigureData } from '@qovery/shared/interfaces'
+import { toastError } from '@qovery/shared/toast'
+import { AppDispatch, RootState } from '@qovery/store'
+import PageSettingsConfigure from '../../ui/page-settings-configure/page-settings-configure'
+
+export function PageSettingsConfigureFeature() {
+  const { applicationId = '', environmentId = '' } = useParams()
+  const methods = useForm<JobConfigureData>()
+
+  const application: JobResponse | undefined = useSelector<RootState, ApplicationEntity | undefined>(
+    (state) => selectApplicationById(state, applicationId),
+    (a, b) => JSON.stringify(a?.id) === JSON.stringify(b?.id)
+  ) as JobResponse | undefined
+
+  const [loading, setLoading] = useState(false)
+
+  const dispatch = useDispatch<AppDispatch>()
+
+  const toasterCallback = () => {
+    if (application) {
+      dispatch(
+        postApplicationActionsRestart({ applicationId, environmentId, serviceType: getServiceType(application) })
+      )
+    }
+  }
+
+  useEffect(() => {
+    methods.setValue('max_duration', application?.max_duration_seconds)
+    methods.setValue('nb_restarts', application?.max_nb_restart)
+    methods.setValue('port', application?.port || undefined)
```

Comment:
I tried changing that with reset, but it breaks my test and does not work at the init strangely. If I go on the settings page, if I refresh on this specific page, it works with the setValue, but does not work with the reset. Unless I add a timeout...

The official doc mention:
> Avoid calling reset before useForm's useEffect is invoked, this is because useForm's subscription needs to be ready before reset can send a signal to flush form state update.
https://react-hook-form.com/api/useform/reset/#main

Maybe it is what happens in this case, the reset is called before the useForm and we never go back in the useEffect after. Gonna try to find a fix for that, otherwise I will keep the setValue.

For information the refactorised code was this one:

```
useEffect(() => {
    const defaultValues: JobConfigureData = {}
    if (application && methods.getValues()) {
      defaultValues.max_duration = application.max_duration_seconds
      defaultValues.nb_restarts = application.max_nb_restart
      defaultValues.port = application.port || undefined

      if (isCronJob(application)) {
        defaultValues.schedule = application.schedule?.cronjob?.scheduled_at || undefined
        defaultValues.cmd_arguments = JSON.stringify(application?.schedule?.cronjob?.arguments) || undefined
        defaultValues.image_entry_point = application.schedule?.cronjob?.entrypoint || undefined
      } else {
        console.log(application)
        defaultValues.on_start = {
          enabled: !!application?.schedule?.on_start,
          arguments_string: JSON.stringify(application?.schedule?.on_start?.arguments),
          entrypoint: application.schedule?.on_start?.entrypoint,
        }

        defaultValues.on_stop = {
          enabled: !!application?.schedule?.on_stop,
          arguments_string: JSON.stringify(application?.schedule?.on_stop?.arguments),
          entrypoint: application.schedule?.on_stop?.entrypoint,
        }

        defaultValues.on_delete = {
          enabled: !!application?.schedule?.on_delete,
          arguments_string: JSON.stringify(application?.schedule?.on_delete?.arguments),
          entrypoint: application.schedule?.on_delete?.entrypoint,
        }
      }
      methods.reset(defaultValues)
    }
  }, [application, methods])
```

Reply:
According to this discussion about the same topic, it seems that for this case where we set the values at the initialisation of our page, it's better to use setValues

https://github.com/react-hook-form/react-hook-form/issues/7753